qiskit/primitives/base/bindings_array.py

```
@@ -0,0 +1,356 @@
+
+        The returned bindings array has a :attr:`shape` given by ``(size, )``, where the size is the
+        :attr:`~size` of this bindings array.
+
+        Returns:
+            A new bindings array.
+        """
+        return self.reshape(self.size)
+
+    def reshape(self, shape: Union[int, Iterable[int]]) -> BindingsArray:
+        """Return a new :class:`~BindingsArray` with a different shape.
+
+        This results in a new view of the same arrays.
+
+        Args:
+            shape: The shape of the returned bindings array.
+
+        Returns:
+            A new bindings array.
+
+        Raises:
+            ValueError: If the provided shape has a different product than the current size.
+        """
+        shape = (shape, -1) if isinstance(shape, int) else (*shape, -1)
+        if np.prod(shape[:-1]).astype(int) != self.size:
+            raise ValueError("Reshaping cannot change the total number of elements.")
+        vals = [val.reshape(shape) for val in self._vals]
+        kwvals = {params: val.reshape(shape) for params, val in self._kwvals.items()}
+        return BindingsArray(vals, kwvals, shape[:-1])
+
+    @classmethod
+    def coerce(cls, bindings_array: BindingsArrayLike) -> BindingsArray:
+        """Coerce BindingsArrayLike into BindingsArray
+
+        Args:
+            bindings_array: an object to be bindings array.
+
+        Returns:
+            A coerced bindings array.
+        """
+        if isinstance(bindings_array, Sequence):
+            bindings_array = np.array(bindings_array)
+        if bindings_array is None:
+            bindings_array = cls([], shape=(1,))
+        elif isinstance(bindings_array, np.ndarray):
+            if bindings_array.ndim == 1:
+                bindings_array = bindings_array.reshape((1, -1))
+            bindings_array = cls(bindings_array)
+        elif isinstance(bindings_array, Mapping):
+            bindings_array = cls(kwargs=bindings_array)
```

```suggestion
            bindings_array = cls(kwvals=bindings_array)
```

---

Should add a final `else` and raise `TypeError` for invalid type

---

Thanks